src/common/resize-event-util/index.js

```diff
@@ -0,0 +1,26 @@
+const handlers = [];
```

Comment:
@yomed I think that would make sense as well.

Also regarding the rational:
` this.subscribeTo(window).on('resize', debounce(() => this.refresh(), 50)); `
Can cause things to jump around as you are resizing the page (which isn't the worst thing but looks kinda ugly). This approach ensures that we are running these resize based changes in line with when when browser renders.

The other benefit to this approach is that the resize event handler is only invoked at most once per animation frame where as the debounced resize function (although it does nothing but defer execution) is actually called on every resize event and this small cost increases as more component instances setup their own handler. This is done by adding the resize handler and immediately removing it once it has been invoked and only adding it again for the next animation frame.

Since a few components need to react to the resize event it seemed to make sense to extract this functionality out into a utility and also address the issue of delayed/debounced updates.

Reply:
Okay I will make the changes

Reply:
@DylanPiercey @sharma46bhawana this is fantastic!

Reply:
@DylanPiercey Smart approach. But one thing to note is that RAF will also fire at the same rate as resize event because it happens before recalc/layout/paint ([explanation](https://bugs.webkit.org/show_bug.cgi?id=177484)). A simple demo that I created [https://resize.glitch.me/](https://resize.glitch.me/). Event count is same in Chrome, whereas Safari RAF count is less because of the bug. 

Having said that your approach still saves on the number of events getting attahced to `resize` itself. At max we will have only one event listner.

Reply:
@senthilp thanks for the link. I did realize that the resize events emitted were in tandem with the   RAF but I had (incorrectly) thought the issue was the other way around and that chrome had the unexpected behaviour and was doing something like it does for `passive` scroll handlers.

Anyway as you say there is still a benefit to this approach, it's just much smaller than I had imagined.
Ultimately though if we decide that this is truly bottlenecking the pages on resize (which doesn't happen too often 😆) we can always update this little utility to revert to the debouncing behaviour while retaining the other benefits.